mono/utils/networking-posix.c

```diff
@@ -127,7 +127,24 @@ mono_get_address_info (const char *hostname, int port, int flags, MonoAddressInf
 
 #endif
 
-#ifdef HAVE_GETPROTOBYNAME
+#ifdef HAVE_GETPROTOBYNAME_R
+
+static int
+fetch_protocol (const char *proto_name, int *cache, int *proto, int default_val)
+{
+	if (!*cache) {
+		struct protoent protoent_buf;
+		struct protoent *pent;
+		char buf[1024];
+
+		getprotobyname_r (proto_name, &protoent_buf, buf, 1024, &pent);
```

Comment:
Check the error? Initialize pent?

Reply:
Error doesn't need to be checked because pent will be set to NULL as per https://linux.die.net/man/3/getprotobyname_r

Reply:
Yuck but ok. I prefer to initialize locals and check errors, and am reluctant to trust documentation when not trusting it is cheap.